src/game.py

```diff
@@ -9,33 +12,40 @@ def initialize_board(self):
         return f"{self.board[0]} | {self.board[1]} | {self.board[2]}\n--+--+--\n{self.board[3]} | {self.board[4]} | {self.board[5]}\n--+--+--\n{self.board[6]} | {self.board[7]} | {self.board[8]}"
 
     def count_plays(self):
-        play_count = self.board.count("X") + self.board.count("O")
+        for mark in self.board:
+            play_count = self.board.count(self.player_one) + self.board.count(
+                self.player_two
+            )
         return play_count
 
-    def get_next_player(self, number):
-        if number == 0:
-            return "X"
-        elif number % 2 == 0:
-            return "X"
+    def get_next_player(self, play_count):
+        if play_count == 0:
+            return self.player_one
+        elif play_count % 2 == 0:
+            return self.player_one
         else:
-            return "O"
+            return self.player_two
 
     def get_prompt(self, play_count):
         next_player = self.get_next_player(play_count)
-        if play_count == 9:
+        if play_count == len(self.board):
             prompt = "Game Over!"
         else:
-            prompt = f"Player {next_player} - enter the number of an empty spot"
+            prompt = f"Player {next_player} - enter a number to place your mark"
         return prompt
 
-    def handle_first_player(self):
-        position_choice = input()
-        position_choice = int(position_choice) - 1
-        self.board[position_choice] = "X"
+    def place_mark_on_board(self, user_input, board, play_count):
+        user_input = user_input - 1
```

Comment:
Can you store this in a new variable to avoid mutating the function argument?

Reply:
fixed - changed variable to `input_index`